x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/rule_execution_logger/logger.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { sum } from 'lodash';
+import { Logger } from 'kibana/server';
+
+import {
+  RuleExecutionStatus,
+  RuleExecutionStatusOrder,
+  RuleExecutionMetrics,
+} from '../../../../../common/detection_engine/schemas/common';
+
+import { RuleAlertType } from '../../rules/types';
+import { ExtMeta } from '../types';
+import { truncateMessage } from '../utils/normalization';
+
+import { IRuleExecutionEventsWriter } from '../rule_execution_events/events_writer';
+import { IRuleExecutionInfoSavedObjectsClient } from '../rule_execution_info/saved_objects_client';
+import { IRuleExecutionLogger, StatusChangeArgs } from './logger_interface';
+
+export const createRuleExecutionLogger = (
+  rule: RuleAlertType,
+  spaceId: string,
+  executionEventsWriter: IRuleExecutionEventsWriter,
+  executionInfoClient: IRuleExecutionInfoSavedObjectsClient,
+  logger: Logger
+): IRuleExecutionLogger => {
+  const ruleId = rule.id;
+  const ruleName = rule.name;
+  const ruleType = rule.alertTypeId;
+
+  const ruleExecutionLogger: IRuleExecutionLogger = {
+    async logStatusChange(args) {
+      try {
+        const normalizedArgs = normalizeStatusChangeArgs(args);
+        await Promise.all([
+          writeStatusChangeToSavedObjects(normalizedArgs),
+          writeStatusChangeToEventLog(normalizedArgs),
+        ]);
+      } catch (e) {
```

Comment:
Yup, thanks for the catch! I added `await` to `writeStatusChangeToSavedObjects` and removed `async` from `writeStatusChangeToEventLog`.

Reply:
Do we still want `writeStatusChangeToEventLog` in the `Promise.all()` now that it's not `async` and doesn't return a `Promise<>`?

IDE calls attention to it: ![image](https://user-images.githubusercontent.com/2946766/150030221-1eac6e9b-8323-4d0d-947c-5e2202f62cf2.png)

Reply:
Technically it doesn't matter since `Promise.all` supports normal values (including `undefined`) in addition to promises. From the code style perspective I'm ok with any variant you prefer. VS Code doesn't show it as an issue, but if WebStorm does, I can change it.